crates/blockchain-tree/src/block_buffer.rs

```diff
@@ -157,14 +155,12 @@ impl BlockBuffer {
     /// The block might be missing from other collections, the method will only ensure that it has
     /// been removed.
     fn remove_block(&mut self, hash: &BlockHash) -> Option<SealedBlockWithSenders> {
-        if let Some(block) = self.blocks.remove(hash) {
+        self.blocks.remove(hash).map(|block| {
             self.remove_from_earliest_blocks(block.number, hash);
             self.remove_from_parent(block.parent_hash, hash);
             self.lru.pop(hash);
-            Some(block)
-        } else {
-            None
-        }
+            block
+        })
```

Comment:
this can be replaced with `?` operator?

Reply:
I prefer this too

Reply:
I'm not sure where you want to use the `?` operator here? We want to return a `None` if the `remove` returns a `None` hence the use of the `map` here. Can you quickly show me where you would like to use this operator?

Reply:
isn't this just

```rust
let block = self.blocks.remove(hash)?;
....
Some(block)
```

Reply:
Ah yes true, just fixed